pkg/prometheus/promcfg.go

```diff
@@ -1854,6 +1859,36 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 		})
 	}
 
+	// Specific configuration generated for Prometheus Agent in DaemonSet mode.
+	if cg.PromAgentDaemonSet {
+		additionalConfigForpromAgentDaemonSet := yaml.MapSlice{
+			{
+				Key:   "role",
+				Value: "pod",
+			},
+			{
+				Key: "selectors",
+				Value: yaml.MapSlice{
+					{
+						Key:   "role",
+						Value: "pod",
+					},
+					{
+						Key:   "field",
+						Value: "spec.nodeName=$(NODE_NAME)",
+					},
+				},
+			},
+		}
```

Comment:
@simonpasquier @ArthurSens The new unit test `TestPromAgentDaemonSetPodMonitorConfig` shows:
```
kubernetes_sd_configs:
          - role: pod
            namespaces:
              names:
              - default
```

It means the new necessary config for daemonset implemented here was not added.

Could you help me figure out why?